src/realm/group.cpp

```diff
@@ -967,6 +967,10 @@ void Group::write(File& file, const char* encryption_key, uint_fast64_t version_
 
     file.set_encryption_key(encryption_key);
 
+    // Force the file system to allocate a node so we get a stable unique id.
+    // See File::get_unique_id(). This is used to distinguish encrypted mappings.
+    file.resize(1);
```

Comment:
shouldn't it be done only for write with the encrypted key?

Reply:
Currently, the bug only manifests for encrypted Realms. But resizing doesn't hurt, and it could prevent a similar issue on unencrypted Realms if future code tries to use the ids in an incorrect way.